### PR TITLE
Upgrade K8s dependencies to v1.29

### PR DESCRIPTION
**Pull request by tenzen-y**
I upgraded the K8s dependencies to v1.29. Also, I upgraded the scheduler-plugins version to the latest (v1.28.9).

### CODE REVIEW

**Comment by tenzen-y** [comment]
/hold
for https://github.com/kubeflow/mpi-operator/pull/631#discussion_r1568320647

**Comment by tenzen-y** [comment]
@alculquicondor The Bumping Go version seems to depend on this PR: https://github.com/kubeflow/mpi-operator/actions/runs/8722386744/job/23928137672?pr=633

So, I'd like to bump the go version in [your PR](https://github.com/kubeflow/mpi-operator/pull/636).

**Comment by alculquicondor** [comment]
can you rebase?

**Comment by tenzen-y** [comment]
> can you rebase?

I have rebased, but I faced other issues. So, I'm trying to investigate the issue.

File: Makefile [hunk 1]
```diff
@@ -197,14 +197,15 @@ scheduler-plugins-chart: scheduler-plugins-crd
 
 .PHONY: volcano-scheduler
 volcano-scheduler:
-	-@GOPATH=/tmp go install volcano.sh/volcano/cmd/scheduler@$(VOLCANO_SCHEDULER_VERSION)
+	rm -rf /tmp/volcano.sh/volcano
+	git clone -b $(VOLCANO_SCHEDULER_VERSION) --depth 1 https://github.com/volcano-sh/volcano /tmp/volcano.sh/volcano
```
Comment (tenzen-y) [inline Makefile:201]: Due to https://github.com/golang/go/issues/44840, `go install` doesn't work fine.

File: Makefile [hunk 2]
```diff
@@ -118,11 +118,11 @@ test_images:
 
 .PHONY: tidy
 tidy:
-	go mod tidy -go 1.20
+	go mod tidy -go 1.22
```
Comment (alculquicondor) [inline Makefile:121]:
```suggestion
	go mod tidy
```

It's already in `go.mod`

File: Makefile [hunk 3]
```diff
@@ -118,11 +118,11 @@ test_images:
...
 
 .PHONY: lint
 lint: bin/golangci-lint ## Run golangci-lint linter
-	$(GOLANGCI_LINT) run --new-from-rev=origin/master --go 1.20
+	$(GOLANGCI_LINT) run --new-from-rev=origin/master --go 1.22
```
Comment (alculquicondor) [inline Makefile:125]: can we also remove this flag?

File: Makefile (hunk 2 above)
Comment (tenzen-y) [reply Makefile]: That makes sense.
Done.

File: Makefile (hunk 3 above)
Comment (tenzen-y) [reply Makefile]: SGTM
Done.

File: test/e2e/e2e_suite_test.go [hunk 4]
```diff
@@ -172,7 +172,7 @@ func bootstrapKindCluster() error {
 }
 
 func installOperator() error {
-	err := runCommand(kubectlPath, "apply", "-k", operatorManifestsPath)
+	err := runCommand(kubectlPath, "apply", "--server-side=true", "-k", operatorManifestsPath)
```
Comment (alculquicondor) [inline test/e2e/e2e_suite_test.go:175]: interesting.... This will have to go to the documentation as well.

Comment (alculquicondor) [reply test/e2e/e2e_suite_test.go:175]: For now, update the README. Just `--server-side` should be enough.

**Comment by tenzen-y** [comment]
The `eventChecker` seems to be broken...

**Comment by tenzen-y** [comment]
> [The `eventChecker`](https://github.com/kubeflow/mpi-operator/blob/a1453cf9a6922fed29cc17b2fe523e7f633e67b7/test/integration/main_test.go#L192) seems to be broken...

NVM. I found another reason.

File: test/integration/main_test.go [hunk 5]
```diff
@@ -180,8 +180,12 @@ func (c *eventChecker) run() {
 		c.Lock()
 		if c.expected.Len() != 0 {
 			front := c.expected.Front()
-			diff := cmp.Diff(front.Value, ev, cmpopts.IgnoreTypes(metav1.ObjectMeta{}), cmpopts.IgnoreFields(corev1.Event{}, "FirstTimestamp", "LastTimestamp", "Count", "Message"), cmpopts.IgnoreFields(corev1.ObjectReference{}, "ResourceVersion"))
-			if diff == "" {
+			cmpOpts := []cmp.Option{
+				cmpopts.IgnoreTypes(metav1.ObjectMeta{}),
+				cmpopts.IgnoreFields(corev1.Event{}, "FirstTimestamp", "LastTimestamp", "Count", "Message", "ReportingController"),
```
Comment (tenzen-y) [inline test/integration/main_test.go:185]: It seems that dedicated `ReportingController` is returned.

```shell
I0418 01:30:35.591211   23178 event.go:364] Event(v1.ObjectReference{Kind:"MPIJob", Namespace:"test-2mcds", Name:"job", UID:"ef9b9814-9220-4d4c-bbdd-e93b23486d6d", APIVersion:"kubeflow.org/v2beta1", ResourceVersion:"223", FieldPath:""}): type: 'Normal' reason: 'MPIJobCreated' MPIJob test-2mcds/job is created.
+++++++++++++++++++++++++++++++++++:   &v1.Event{
        ... // 5 ignored and 7 identical fields
        Action:              "",
        Related:             nil,
-       ReportingController: "",
+       ReportingController: "mpi-job-controller",
        ReportingInstance:   "",
  }
```

File: test/integration/mpi_job_controller_test.go [hunk 6]
```diff
@@ -1176,6 +1176,7 @@ func newInt32(v int32) *int32 {
 func eventForJob(event corev1.Event, job *kubeflow.MPIJob) corev1.Event {
 	event.Namespace = job.Namespace
 	event.Source.Component = "mpi-job-controller"
+	event.ReportingController = "mpi-job-controller"
```
Comment (tenzen-y) [inline test/integration/mpi_job_controller_test.go:1179]: It seems that dedicated `ReportingController` is returned.

```shell
I0418 01:30:35.591211   23178 event.go:364] Event(v1.ObjectReference{Kind:"MPIJob", Namespace:"test-2mcds", Name:"job", UID:"ef9b9814-9220-4d4c-bbdd-e93b23486d6d", APIVersion:"kubeflow.org/v2beta1", ResourceVersion:"223", FieldPath:""}): type: 'Normal' reason: 'MPIJobCreated' MPIJob test-2mcds/job is created.
+++++++++++++++++++++++++++++++++++:   &v1.Event{
        ... // 5 ignored and 7 identical fields
        Action:              "",
        Related:             nil,
-       ReportingController: "",
+       ReportingController: "mpi-job-controller",
        ReportingInstance:   "",
  }
```

**Comment by tenzen-y** [comment]
@alculquicondor Could you take another look?

File: .github/workflows/golangci-lint.yml [hunk 7]
```diff
@@ -18,6 +18,6 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.53.3
-          args: --timeout=5m --go=1.20
+          version: v1.57.2
+          args: --timeout=5m --go=1.22
```
Comment (alculquicondor) [inline .github/workflows/golangci-lint.yml:22]: can we remove `--go=1.22`?

**Review by alculquicondor — COMMENTED** [review]
just one nit

File: .github/workflows/golangci-lint.yml (hunk 7 above)
Comment (tenzen-y) [reply .github/workflows/golangci-lint.yml]: Nice catch!
Done.

**Pull request merged**